src/components/ContentNode/Grid.vue

```diff
@@ -0,0 +1,55 @@
+<!--
+  This source file is part of the Swift.org open source project
+
+  Copyright (c) 2022 Apple Inc. and the Swift project authors
+  Licensed under Apache License v2.0 with Runtime Library Exception
+
+  See https://swift.org/LICENSE.txt for license information
+  See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+-->
+<template>
+  <div class="grid" :style="style" :class="{ 'with-columns': columns }">
+    <slot />
+  </div>
+</template>
+
+<script>
+export default {
+  name: 'grid',
+  props: {
+    columns: {
+      type: Number,
+      default: null,
+      required: false,
+      validator: v => v > 0,
+    },
+  },
+  computed: {
+    style: ({ columns }) => ({ '--col-count': columns }),
```

Comment:
What happens when `columns` is `null` here? Should there be a default value of `12` since that is what used in the `grid-template-columns`  when this var isn't defined anyways?

Reply:
Actually this is a good question. That 12 is never used when I think about it.

My idea was that a Grid by default has no column count, so every item you add in there, will take 1fr. It might make sense to add a limit to that though, so things wrap after a certain point 🤔 

Then if you specify a column count, it splits the content into those columns, so you can even have 2x2 and 1x1 on a 5 col grid.

I will look into it again, and see if I can add a limit.

Reply:
OK I did some changes. Let me know how that looks.

Reply:
It looks very similar to before. So what happens when it's `null`? Should `12` just be the default since that's the value I still see being used in the CSS when this prop isn't provided?

Reply:
I did a small refactor. We need a way to distinguish between a pre-defined N column grid, vs one that accepts arbitrary columns, up to a max of 12.

Let me know if it looks OK the way I have it now.